src/components/modal/v2/lib/zoid-polyfill.js

```diff
@@ -1,15 +1,46 @@
 /* global Android */
 import { isAndroidWebview, isIosWebview, getPerformance } from '@krakenjs/belter/src';
 import { getOrCreateDeviceID, logger } from '../../../../utils';
-import { isIframe } from './utils';
+import { isIframe, validateProps, sendEventAck } from './utils';
 
 const IOS_INTERFACE_NAME = 'paypalMessageModalCallbackHandler';
 const ANDROID_INTERFACE_NAME = 'paypalMessageModalCallbackHandler';
 
+function listenAndAssignProps(newProps, propListeners) {
+    Array.from(propListeners.values()).forEach(listener => {
+        listener({ ...window.xprops, ...newProps });
+    });
+    Object.assign(window.xprops, newProps);
+}
+
+export function validateAndUpdateBrowserProps(initialProps, propListeners, updatedPropsEvent) {
+    const {
+        origin: eventOrigin,
+        data: { eventName, id, eventPayload: newProps }
+    } = updatedPropsEvent;
+    const merchantOrigin = decodeURIComponent(initialProps.origin);
+
+    if (eventOrigin === merchantOrigin && eventName === 'PROPS_UPDATE' && newProps && typeof newProps === 'object') {
+        // send event ack so PostMessenger will stop reposting event
+        sendEventAck(id, merchantOrigin);
```

Comment:
I know we chatted about this internally, but I think we can go with 
```suggestion
    const clientOrigin = decodeURIComponent(initialProps.origin);

    if (eventOrigin === clientOrigin && eventName === 'PROPS_UPDATE' && newProps && typeof newProps === 'object') {
        // send event ack so PostMessenger will stop reposting event
        sendEventAck(id, clientOrigin);
```